gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/oauth/OAuthViewModel.kt

```diff
@@ -0,0 +1,69 @@
+package com.gravatar.quickeditor.ui.oauth
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.viewmodel.CreationExtras
+import com.gravatar.quickeditor.data.service.WordPressOAuthService
+import com.gravatar.services.Result
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+internal class OAuthViewModel(
+    private val wordPressOAuthService: WordPressOAuthService,
+) : ViewModel() {
+    private val _uiState = MutableStateFlow(OAuthUiState())
+    val uiState: StateFlow<OAuthUiState> = _uiState.asStateFlow()
+
+    private val _actions = Channel<OAuthAction>(Channel.BUFFERED)
+    val actions = _actions.receiveAsFlow()
+
+    init {
+        viewModelScope.launch {
+            _actions.send(OAuthAction.StartOAuth)
+        }
+    }
```

Comment:
This is possible thanks to `navigation-compose`. Otherwise, we would always get the same `ViewModel` instance, so doing anything in onInit would make sense (the same instance on each QE launch)